default.nix

```diff
@@ -123,6 +123,7 @@ let haskellPackages =
             else [];
         # Use a better prompt
         shellHook = ''
+          export CABAL_DIR="$(pwd)/.cabal.nix"
```

Comment:
> The CABAL_DIR might be dropped in the future, when cabal-install starts to use XDG Directory specification.

https://cabal.readthedocs.io/en/3.4/installing-packages.html

Reply:
That would be even better then we will have the XDG environment variables to use see https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html .

Reply:
What is `.cabal.nix`? How does `CABAL_DIR` understand that?

Reply:
It is just a location for cabal to store its config/store, non-existing dir to begin with, when you invoke cabal it will create its stuff there. CABAL_DIR is a way to tell that location to cabal.